src/Extension.ts

```diff
@@ -45,6 +45,44 @@ export class Extension {
 	);
 
 	constructor(private readonly context: vscode.ExtensionContext) {
+		this.dispose.track(
+			vscode.commands.registerCommand(
+				"hediet.vscode-drawio.newDiagram",
+				async () => {
+					const newFileName = await vscode.window.showInputBox({
+						placeHolder: "Enter file name",
+					});
+
+					if (newFileName) {
+						// Figure the folder to put new file
+						const openFilePath =
+							vscode.window.activeTextEditor?.document.uri.path;
```

Comment:
This is really confusing behaviour! The user should be aware of any implicit assumption the UI does.